python/ray/experimental/channel/torch_tensor_nccl_channel.py

```diff
@@ -25,6 +26,118 @@
 logger = logging.getLogger(__name__)
 
 
+class NestedTorchTensorNcclChannel(ChannelInterface):
+    def __init__(
+        self,
+        writer: ray.actor.ActorHandle,
+        readers: List[ray.actor.ActorHandle],
+        gpu_data_typ: "TorchTensorType",
+        cpu_data_typ: Optional["SharedMemoryType"] = None,
+        _gpu_data_channel: Optional["TorchTensorNcclChannel"] = None,
+        _cpu_data_channel: Optional["Channel"] = None,
+    ):
+        self._writer = writer
+        self._readers = readers
+
+        if _gpu_data_channel is not None or _cpu_data_channel is not None:
```

Comment:
It is possible for `_gpu_data_channel` to be not None, `_cpu_data_channel` to be None, and `cpu_data_typ` to be not None.

Reply:
Added assertion checks for this.